tests/test_qt_plotting.py

```diff
@@ -426,7 +457,6 @@ def test_background_plotting_close(qtbot, close_event, empty_scene):
 
     # check that BackgroundPlotter.__init__() is called
     assert _hasattr(plotter, "app_window", MainWindow)
-    assert _hasattr(plotter, "main_menu", QMenuBar)
```

Comment:
I assume this is connected to the comment about out of context checks? I don't think it's ever bad to have additional checks - maybe there's some weird interaction that noone has thought of but might still become relevant "in the wild".

Reply:
This test doesn't create the `main_menu`, and anyway, `main_menu` would be `None` since we're assigning the attribute.  We could test for `main_menu` is `None` here, but the rest of the tests would be out of context.

Reply:
Actually it does create the main menu since it's the default: `main_menu=True`

Reply:
Ah, sorry!  We should probably include the asserts then.